client/src/components/FavoriteCoins.js

```diff
@@ -0,0 +1,141 @@
+import styled from 'styled-components/macro';
+import PropTypes from 'prop-types';
+import favoriteIcon from '../image/star.png';
+
+export default function FavoriteCoinsCard({
+  index,
+  topCoin,
+  onToggleFavoriteCoins,
+  onSetSelectedCoin,
+  onSetWalletOverview,
+  walletOverview,
+}) {
+  function ShortenLargeNumbers(num) {
+    if (num > 999999999) {
+      return Math.sign(num) * (Math.abs(num) / 1000000000).toFixed(1) + ' B';
+    } else {
+      return Math.sign(num) * (Math.abs(num) / 1000000).toFixed(1) + ' M';
+    }
+  }
+  return (
+    <>
+      <CoinWrapper key={index + topCoin}>
+        <section>
+          <CoinNameSymbole>
+            <CoinImage src={topCoin.image} alt="Image of Coin" />
+            <span>
+              {topCoin.name}{' '}
+              <FavoriteIcon
+                src={favoriteIcon}
+                alt="Star image of icon"
+                onClick={() => onToggleFavoriteCoins(topCoin)}
+                style={
+                  topCoin.isFavorite ? { opacity: '100%' } : { opacity: '30%' }
```

Comment:
consider removing the inline styles here as well and move them to the styled components section :)